app/controllers/devise_sessions_controller.rb

```diff
@@ -3,54 +3,81 @@ class DeviseSessionsController < Devise::SessionsController
   include CookiesHelper
 
   before_action :check_login_state, only: %i[
-    create
     phone_code
-    phone_code_send
     phone_verify
     phone_resend
   ]
 
   before_action :check_password_ok, only: %i[
     phone_code
-    phone_code_send
     phone_verify
     phone_resend
   ]
 
   def create
```

Comment:
Do we need `skip_before_action :verify_authenticity_token, only: [:create]` in this controller for finder-frontend to be able to POST the JWT?

Reply:
Oh, I see, it's still sent to the welcome controller, but passed in a param to this controller.

I'm not sure that will work with large JWTs as there's a limit on URL size, which is why we have the POST thing in the first place.

Reply:
I'll add the skip authenticity token to this controller.

I think there's a general issue with the JWT and splitting the login/registration flows: if users come to the registration form, but want to login, we'll need to POST the token onto the login form.  The current designs use a link, rather than button to move between the forms.  I'm not sure we could use a cookie to store this, as the JWT might not be considered essential?

Reply:
One way around that would be merging the `LoginState` and `RegistrationState` types so it doesn't matter which controller you originally arrive in.

Reply:
I've kept them separate, but created a new table for storing the JWTs, with the id persisted in the session (in the same way as we do for the registration and login flows).

The last commit is still WIP (just pushing it because I'm logging off for the day) so probably not worth reviewing yet as there will still be some missing parts.